SideVM compiling error on Apple Silicon

```
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-Wl,-exported_symbols_list,/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/rustcGCH7p0/list" "-arch" "arm64" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.0.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.1.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.10.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.11.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.12.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.2.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.3.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.4.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.5.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.6.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.7.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.8.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.sidevm_recv_messages.38be40f8-cgu.9.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_recv_messages.4kkfvf1kz649zr8.rcgu.o" "-L" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps" "-L" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm-eada62ab3d78cfa6.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhyper-96ee29bafb681ef9.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libwant-2f2ceaba4437c7ca.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtry_lock-2fdbe0a5e72d4a82.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttparse-c4b49e9599558849.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libh2-9af4e29ab8bdb359.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libindexmap-e8f516856afe56d8.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhashbrown-ed0e4eaf579a6028.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libahash-ce7a419fb4c0def4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libgetrandom-755e0d4a66d67cbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtokio_util-7c2976d9191ab9de.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtower_service-6c60fc440cbdadd4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtracing-3515c1d9e7b91b5a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtracing_core-17b01b93a91a0fd4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblazy_static-ebd75cc0f36ef66d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttp_body-a26c993d00cdc7a2.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_util-82e362e929cc7056.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures-14536ffb6cf9225f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_io-2e47f1c11c0acecb.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libslab-c069afdb76c7c800.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_channel-c5176b75985f2cef.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_sink-e050bc015963b534.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_task-525fda40fd83554f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpin_utils-f629fb86d1cbfb4d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_core-77c4ee147c1c7cd5.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtokio-627c24925a7f8310.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsignal_hook_registry-413bca2417cf1687.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libnum_cpus-099930b8d025c3af.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsocket2-f9e200494d2198a1.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libmemchr-d0e4df78ed70cea4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libonce_cell-544004bdff825ddf.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libmio-f2663237c204a7ad.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparking_lot-7537af37618732b0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparking_lot_core-cf9d40dc64199bc7.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblibc-622458a9597338ee.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsmallvec-df6870849ccfc262.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblock_api-8b485743796c08c0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libscopeguard-15e20c72e8a90004.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpin_project_lite-329566b13e6e45f0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttpdate-ffd923ada5337cbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttp-729942fcedac7b29.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libitoa-5f8373da6ca63a9c.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbytes-63d05a2d41c7e73a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfnv-0419508d61057d71.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm_logger-8f6ff2e73acedd61.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm_env-385df31833fa7fbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtinyvec-578cef4b7f7be158.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtinyvec_macros-b04713a4a486580f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparity_scale_codec-56a571e158eace37.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libserde-a009ad3d106027f9.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbitvec-4369d5f585ae8e0c.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libwyz-4788476a10450a79.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtap-1df9a2073a5d8394.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libradium-0285603de30e8e39.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfunty-8ea5e994075d5b24.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libarrayvec-03da2c75b92c6a4a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbyte_slice_cast-e12e423a0331350d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libnum_enum-6e711b44697fcc74.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblog-8e38667df84f4033.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libvalue_bag-fb9b0912090ca6a7.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libcfg_if-0692cee5d0087e61.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd-65593617647a200e.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libpanic_unwind-8d961ef8c4f63984.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libobject-899dcf63b86be225.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libmemchr-ac8c24923a4ceb78.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libaddr2line-4c2771a15fb6f37e.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libgimli-4e9627c2a071def0.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_demangle-7939dcc7209ec7df.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd_detect-ff2bce2657742dae.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libhashbrown-5d05867f34d20949.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libminiz_oxide-0c954423be007412.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libadler-02b3d63b95c699db.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_alloc-1598edcd5bca28f3.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libunwind-96d716517eaa7162.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcfg_if-0dd17ccd4131f308.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liblibc-53fbf10f6e15a5b4.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liballoc-852ad786540c5430.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_core-95a7e52c78a3fba1.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcore-dd6de681850a671c.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-68b99a601d9fe98d.rlib" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsidevm_recv_messages.dylib" "-Wl,-dead_strip" "-dynamiclib" "-Wl,-dylib" "-nodefaultlibs"
  = note: Undefined symbols for architecture arm64:
            "_sidevm_ocall", referenced from:
                pink_sidevm_env::ocall_def::ocall_funcs_guest::poll::hfcb51dcd40e4aebe in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
            "_sidevm_ocall_fast_return", referenced from:
                pink_sidevm_env::ocall_def::ocall_funcs_guest::poll::hfcb51dcd40e4aebe in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::mark_task_ready::hbb03792467d2dff7 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::next_ready_task::h3e2def7115dc63ce in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::enable_ocall_trace::h214eb1cb93982e8b in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::log::hff19b337e34eab03 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
          ld: symbol(s) not found for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: could not compile `sidevm-recv-messages` due to previous error
warning: build failed, waiting for other jobs to finish...
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-Wl,-exported_symbols_list,/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/rustcj5Bn9g/list" "-arch" "arm64" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.0.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.1.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.10.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.11.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.12.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.13.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.14.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.15.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.2.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.3.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.4.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.5.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.6.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.7.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.8.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.sidevm_timer.530b2ee7-cgu.9.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_timer.31s0q0aunxepjghd.rcgu.o" "-L" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps" "-L" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm-eada62ab3d78cfa6.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhyper-96ee29bafb681ef9.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libwant-2f2ceaba4437c7ca.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtry_lock-2fdbe0a5e72d4a82.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttparse-c4b49e9599558849.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libh2-9af4e29ab8bdb359.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libindexmap-e8f516856afe56d8.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhashbrown-ed0e4eaf579a6028.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libahash-ce7a419fb4c0def4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libgetrandom-755e0d4a66d67cbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtokio_util-7c2976d9191ab9de.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtower_service-6c60fc440cbdadd4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtracing-3515c1d9e7b91b5a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtracing_core-17b01b93a91a0fd4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblazy_static-ebd75cc0f36ef66d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttp_body-a26c993d00cdc7a2.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_util-82e362e929cc7056.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures-14536ffb6cf9225f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_io-2e47f1c11c0acecb.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libslab-c069afdb76c7c800.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_channel-c5176b75985f2cef.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_sink-e050bc015963b534.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_task-525fda40fd83554f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpin_utils-f629fb86d1cbfb4d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_core-77c4ee147c1c7cd5.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtokio-627c24925a7f8310.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsignal_hook_registry-413bca2417cf1687.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libnum_cpus-099930b8d025c3af.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsocket2-f9e200494d2198a1.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libmemchr-d0e4df78ed70cea4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libonce_cell-544004bdff825ddf.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libmio-f2663237c204a7ad.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparking_lot-7537af37618732b0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparking_lot_core-cf9d40dc64199bc7.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblibc-622458a9597338ee.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsmallvec-df6870849ccfc262.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblock_api-8b485743796c08c0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libscopeguard-15e20c72e8a90004.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpin_project_lite-329566b13e6e45f0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttpdate-ffd923ada5337cbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttp-729942fcedac7b29.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libitoa-5f8373da6ca63a9c.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbytes-63d05a2d41c7e73a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfnv-0419508d61057d71.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm_logger-8f6ff2e73acedd61.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm_env-385df31833fa7fbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtinyvec-578cef4b7f7be158.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtinyvec_macros-b04713a4a486580f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparity_scale_codec-56a571e158eace37.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libserde-a009ad3d106027f9.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbitvec-4369d5f585ae8e0c.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libwyz-4788476a10450a79.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtap-1df9a2073a5d8394.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libradium-0285603de30e8e39.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfunty-8ea5e994075d5b24.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libarrayvec-03da2c75b92c6a4a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbyte_slice_cast-e12e423a0331350d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libnum_enum-6e711b44697fcc74.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblog-8e38667df84f4033.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libvalue_bag-fb9b0912090ca6a7.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libcfg_if-0692cee5d0087e61.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd-65593617647a200e.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libpanic_unwind-8d961ef8c4f63984.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libobject-899dcf63b86be225.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libmemchr-ac8c24923a4ceb78.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libaddr2line-4c2771a15fb6f37e.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libgimli-4e9627c2a071def0.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_demangle-7939dcc7209ec7df.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd_detect-ff2bce2657742dae.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libhashbrown-5d05867f34d20949.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libminiz_oxide-0c954423be007412.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libadler-02b3d63b95c699db.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_alloc-1598edcd5bca28f3.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libunwind-96d716517eaa7162.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcfg_if-0dd17ccd4131f308.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liblibc-53fbf10f6e15a5b4.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liballoc-852ad786540c5430.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_core-95a7e52c78a3fba1.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcore-dd6de681850a671c.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-68b99a601d9fe98d.rlib" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsidevm_timer.dylib" "-Wl,-dead_strip" "-dynamiclib" "-Wl,-dylib" "-nodefaultlibs"
  = note: Undefined symbols for architecture arm64:
            "_sidevm_ocall_fast_return", referenced from:
                pink_sidevm_env::ocall_def::ocall_funcs_guest::close::h005c4ea796c573fd in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::poll_read::hbb03df2da470c691 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::mark_task_ready::hbb03792467d2dff7 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::next_ready_task::h3e2def7115dc63ce in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::enable_ocall_trace::h214eb1cb93982e8b in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::create_timer::h57984d6e56878ce4 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::log::hff19b337e34eab03 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                ...
          ld: symbol(s) not found for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: could not compile `sidevm-timer` due to previous error
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-Wl,-exported_symbols_list,/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/rustcXdXHZQ/list" "-arch" "arm64" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.0.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.1.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.10.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.11.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.12.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.13.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.14.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.15.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.2.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.3.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.4.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.5.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.6.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.7.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.8.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.sidevm_httpserver.99b6dbdc-cgu.9.rcgu.o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/sidevm_httpserver.pdkdz6hidjjci8p.rcgu.o" "-L" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps" "-L" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm-eada62ab3d78cfa6.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhyper-96ee29bafb681ef9.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libwant-2f2ceaba4437c7ca.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtry_lock-2fdbe0a5e72d4a82.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttparse-c4b49e9599558849.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libh2-9af4e29ab8bdb359.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libindexmap-e8f516856afe56d8.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhashbrown-ed0e4eaf579a6028.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libahash-ce7a419fb4c0def4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libgetrandom-755e0d4a66d67cbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtokio_util-7c2976d9191ab9de.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtower_service-6c60fc440cbdadd4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtracing-3515c1d9e7b91b5a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtracing_core-17b01b93a91a0fd4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblazy_static-ebd75cc0f36ef66d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttp_body-a26c993d00cdc7a2.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_util-82e362e929cc7056.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures-14536ffb6cf9225f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_io-2e47f1c11c0acecb.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libslab-c069afdb76c7c800.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_channel-c5176b75985f2cef.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_sink-e050bc015963b534.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_task-525fda40fd83554f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpin_utils-f629fb86d1cbfb4d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfutures_core-77c4ee147c1c7cd5.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtokio-627c24925a7f8310.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsignal_hook_registry-413bca2417cf1687.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libnum_cpus-099930b8d025c3af.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsocket2-f9e200494d2198a1.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libmemchr-d0e4df78ed70cea4.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libonce_cell-544004bdff825ddf.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libmio-f2663237c204a7ad.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparking_lot-7537af37618732b0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparking_lot_core-cf9d40dc64199bc7.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblibc-622458a9597338ee.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsmallvec-df6870849ccfc262.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblock_api-8b485743796c08c0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libscopeguard-15e20c72e8a90004.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpin_project_lite-329566b13e6e45f0.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttpdate-ffd923ada5337cbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libhttp-729942fcedac7b29.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libitoa-5f8373da6ca63a9c.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbytes-63d05a2d41c7e73a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfnv-0419508d61057d71.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm_logger-8f6ff2e73acedd61.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libpink_sidevm_env-385df31833fa7fbe.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtinyvec-578cef4b7f7be158.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtinyvec_macros-b04713a4a486580f.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libparity_scale_codec-56a571e158eace37.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libserde-a009ad3d106027f9.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbitvec-4369d5f585ae8e0c.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libwyz-4788476a10450a79.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libtap-1df9a2073a5d8394.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libradium-0285603de30e8e39.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libfunty-8ea5e994075d5b24.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libarrayvec-03da2c75b92c6a4a.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libbyte_slice_cast-e12e423a0331350d.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libnum_enum-6e711b44697fcc74.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/liblog-8e38667df84f4033.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libvalue_bag-fb9b0912090ca6a7.rlib" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libcfg_if-0692cee5d0087e61.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd-65593617647a200e.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libpanic_unwind-8d961ef8c4f63984.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libobject-899dcf63b86be225.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libmemchr-ac8c24923a4ceb78.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libaddr2line-4c2771a15fb6f37e.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libgimli-4e9627c2a071def0.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_demangle-7939dcc7209ec7df.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd_detect-ff2bce2657742dae.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libhashbrown-5d05867f34d20949.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libminiz_oxide-0c954423be007412.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libadler-02b3d63b95c699db.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_alloc-1598edcd5bca28f3.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libunwind-96d716517eaa7162.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcfg_if-0dd17ccd4131f308.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liblibc-53fbf10f6e15a5b4.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liballoc-852ad786540c5430.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_core-95a7e52c78a3fba1.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcore-dd6de681850a671c.rlib" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-68b99a601d9fe98d.rlib" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/jasl/.rustup/toolchains/nightly-2022-04-01-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/Users/jasl/Workspaces/BlockChain/phala-blockchain/target/release/deps/libsidevm_httpserver.dylib" "-Wl,-dead_strip" "-dynamiclib" "-Wl,-dylib" "-nodefaultlibs"
  = note: Undefined symbols for architecture arm64:
            "_sidevm_ocall", referenced from:
                pink_sidevm_env::ocall_def::ocall_funcs_guest::tcp_accept::h454dd96435ed22e0 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
            "_sidevm_ocall_fast_return", referenced from:
                pink_sidevm_env::ocall_def::ocall_funcs_guest::close::h005c4ea796c573fd in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::poll_read::hbb03df2da470c691 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::poll_write::h98d40e24795f90ed in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::mark_task_ready::hbb03792467d2dff7 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::next_ready_task::h3e2def7115dc63ce in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::enable_ocall_trace::h214eb1cb93982e8b in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                pink_sidevm_env::ocall_def::ocall_funcs_guest::tcp_listen::hb48ba804be1f3710 in libpink_sidevm_env-385df31833fa7fbe.rlib(pink_sidevm_env-385df31833fa7fbe.pink_sidevm_env.7ba5e417-cgu.5.rcgu.o)
                ...
          ld: symbol(s) not found for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: could not compile `sidevm-httpserver` due to previous error
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,6 +29,7 @@ exclude = [
 	"native-nostd-hasher",
 	"standalone/pruntime",
 	"crates/pink/examples",
+	"crates/pink/sidevm/examples",
 ]
 
 members = [
@@ -62,10 +63,6 @@ members = [
 	"crates/pink/sidevm/macro",
 	"crates/pink/sidevm/logger",
 	"crates/pink/sidevm/sidevm",
-	"crates/pink/sidevm/examples/timer",
-	"crates/pink/sidevm/examples/httpserver",
-	"crates/pink/sidevm/examples/host",
-	"crates/pink/sidevm/examples/recv_messages",
 	"crates/phala-serde-more",
 	"crates/rustfmt-snippet",
 	"pallets/phala",

--- a/crates/pink/sidevm/examples/host/Cargo.toml
+++ b/crates/pink/sidevm/examples/host/Cargo.toml
@@ -1,3 +1,6 @@
+# workaround for https://github.com/rust-lang/cargo/issues/6745
+[workspace]
+
 [package]
 edition = "2021"
 name = "sidevm-host"

--- a/crates/pink/sidevm/examples/httpserver/Cargo.toml
+++ b/crates/pink/sidevm/examples/httpserver/Cargo.toml
@@ -1,3 +1,6 @@
+# workaround for https://github.com/rust-lang/cargo/issues/6745
+[workspace]
+
 [package]
 edition = "2021"
 name = "sidevm-httpserver"

--- a/crates/pink/sidevm/examples/recv_messages/Cargo.toml
+++ b/crates/pink/sidevm/examples/recv_messages/Cargo.toml
@@ -1,3 +1,6 @@
+# workaround for https://github.com/rust-lang/cargo/issues/6745
+[workspace]
+
 [package]
 edition = "2021"
 name = "sidevm-recv-messages"

--- a/crates/pink/sidevm/examples/timer/Cargo.toml
+++ b/crates/pink/sidevm/examples/timer/Cargo.toml
@@ -1,3 +1,6 @@
+# workaround for https://github.com/rust-lang/cargo/issues/6745
+[workspace]
+
 [package]
 edition = "2021"
 name = "sidevm-timer"

